packages/itwinui-react/src/core/Tile/Tile.tsx

```diff
@@ -147,239 +105,463 @@ type TileProps = {
-                    styleType='borderless'
-                    size='small'
-                    aria-label='More options'
-                  >
-                    <SvgMore />
-                  </IconButton>
-                </Box>
-              </DropdownMenu>
-            )}
-
-            {children}
-          </Box>
-
-          {buttons && <Box className='iui-tile-buttons'>{buttons}</Box>}
-        </Box>
-      </TileContext.Provider>
+      <Box
+        className={cx('iui-tile-thumbnail-picture', className)}
+        style={{
+          backgroundImage:
+            url && typeof url === 'string' ? `url(${url})` : undefined,
+        }}
+        ref={ref}
+        {...rest}
+      />
     );
-  }) as PolymorphicForwardRefComponent<'div', TileProps>,
-  {
-    /**
-     * Polymorphic Tile action component. Recommended to be used in a "name" of `Tile`.
-     * Renders `a` element by default.
-     * @example
-     * <Tile
-     *   name={<Tile.Action href='/new-page'>Tile name<Tile.Action/>}
-     * />
-     */
-    Action: TileAction,
-  },
+  }
+
+  return (
+    <Box className={cx('iui-thumbnail-icon', className)} ref={ref} {...rest}>
+      {children}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'div', TileThumbnailPictureOwnProps>;
+
+TileThumbnailPicture.displayName = 'Tile.TileThumbnailPicture';
+
+// ----------------------------------------------------------------------------
+// Tile.QuickAction component
+
+const TileQuickAction = polymorphic.div('iui-tile-thumbnail-quick-action');
+TileQuickAction.displayName = 'Tile.QuickAction';
+
+// ----------------------------------------------------------------------------
+// Tile.TypeIndicator component
+
+const TileTypeIndicator = polymorphic.div('iui-tile-thumbnail-type-indicator');
+TileTypeIndicator.displayName = 'Tile.TypeIndicator';
+
+// ----------------------------------------------------------------------------
+// Tile.IconButton component
+
+const TileIconButton = React.forwardRef((props, ref) => {
+  const { className, children, ...rest } = props;
+  return (
+    <IconButton
+      className={className}
+      styleType='borderless'
+      size='small'
+      ref={ref}
+      {...rest}
+    >
+      {children}
+    </IconButton>
+  );
+}) as PolymorphicForwardRefComponent<
+  'button',
+  React.ComponentPropsWithoutRef<typeof IconButton>
+>;
+TileIconButton.displayName = 'Tile.IconButton';
+
+// ----------------------------------------------------------------------------
+// Tile.BadgeContainer component
+
+const TileBadgeContainer = polymorphic.div(
+  'iui-tile-thumbnail-badge-container',
 );
+TileBadgeContainer.displayName = 'Tile.BadgeContainer';
 
-type TitleIconProps = {
-  isLoading?: boolean;
-  isSelected?: boolean;
-  isNew?: boolean;
-  status?: 'positive' | 'warning' | 'negative';
+// ----------------------------------------------------------------------------
+// Tile.Name component
+type TileNameOwnProps = {
+  name?: string;
```

Comment:
this is done for convenience so the user doesn't always have to manually pass `TileNameIcon`

Reply:
hmm actually wouldn't having the string as children achieve the same result as having the prop though?

Reply:
`children` is currently rendered as-is. That means `<Tile.Name>some string</Tile.Name>` doesn't render an icon. I think we want to preserve that behavior, otherwise it would be confusing why `<Tile.Name><>string in a fragment</></Tile.Name>` suddenly starts adding an icon

Reply:
on that note, you can make this a react node
```suggestion
  name?: React.ReactNode;
```